torch/nn/modules/module.py

```diff
@@ -1585,6 +1589,16 @@ def _register_state_dict_hook(self, hook):
         self._state_dict_hooks[handle.id] = hook
         return handle
 
+    def register_state_dict_pre_hook(self, hook):
```

Comment:
Curious: Do we want this to be public? If so, is there a reason why `_register_state_dict_hook()` is private?

Reply:
We do want to make these properly public. The issue linked in the description should give details about that!